Сборка openwrt репозитория на github-pages

Vantuz opened the pull request:
Вдохновлялся https://github.com/Anonym-tsk/nfqws-keenetic

Для сборки нужно задать 4 секрета в настройках репы
```
OPENWRT_PUBLIC_KEY
OPENWRT_SECRET_KEY
OPENWRT_APK_PUBLIC_KEY
OPENWRT_APK_SECRET_KEY
```
Как это сделать описано в keys/README

Тестировал на xiaomi r3g c openwrt 24.10.1, архитектура `mipsel_24kc`
Также протестировал на xiaomi ax300t c openwrt 24.10.1, архитектура `aarch64_cortex-a53`

### Установка репозитория на OpenWRT (скопировано у nfqws-keenetic)

#### До версии 24.10 включительно, пакетный менеджер `opkg`

1. Установите необходимые зависимости
   ```
   opkg update
   opkg install ca-certificates wget-ssl
   opkg remove wget-nossl
   ```

2. Установите публичный ключ репозитория
   ```
   wget -O "/tmp/zapret-openwrt.pub" "https://raw.githubusercontent.com/Vantuz/zapret-openwrt/refs/heads/master/keys/public.key"
   opkg-key add /tmp/zapret-openwrt.pub
   ```

3. Установите репозиторий в систему
   ```
   echo "src/gz zapret-openwrt https://vantuz.github.io/zapret-openwrt/repo/ipk/aarch64_cortex-a53" >> /etc/opkg/customfeeds.conf
   ```
   Заменить `aarch64_cortex-a53` на нужную архитектуру

4. Установите пакет
   ```
   opkg update
   opkg install luci-app-zapret
   ```

#### Версии 25.xx и Snapshot, пакетный менеджер `apk`

Эту инструкцию не проверял, могут быть неточности

1. Установите необходимые зависимости
   ```
   apk --update-cache add ca-certificates wget-ssl
   apk del wget-nossl
   ```

2. Установите публичный ключ репозитория
   ```
   wget -O "/etc/apk/keys/zapret-openwrt.pem" "https://raw.githubusercontent.com/Vantuz/zapret-openwrt/refs/heads/master/keys/public-key.pem"
   ```

3. Установите репозиторий в систему
   ```
   echo "https://vantuz.github.io/zapret-openwrt/repo/apk/aarch64_cortex-a53/packages.adb" > /etc/apk/repositories.d/zapret-openwrt.list
   ```
   Заменить `aarch64_cortex-a53` на нужную архитектуру

4. Установите пакет
   ```
   apk --update-cache add luci-app-zapret
   ```

Vantuz commented:
@remittor review plz)

remittor commented:
Так я в курсе всего этого. И тоже смотрел как это сделано в именно в [nfqws-keenetic](https://github.com/Anonym-tsk/nfqws-keenetic)
Да вот только какой профит от этого для пользователя?
Пользователь всё равно должен ручками править `customfeeds.conf`, либо добавлять `adb`. Единственное, что это нужно делать единожды.
Вот вы бы лучше сделали аналог `fantastic-packages` куда поместили бы zapret-openwrt, nfqws-keenetic, youtubeUnblock, podkop, ruantiblock, etc.
Тогда есть смысл заставлять пользователя править `customfeeds.conf`, т.к. профит существенный.

Vantuz commented:
@remittor 
> Единственное, что это нужно делать единожды.

Мне кажется это уже неплохой профит) Потом можно обновлять запрет одной командой или даже кнопкой в люси -- гораздо удобнее чем пакет искать на гитхабе и руками качать. Особенно это удобно после sysupgrade -- можно даже команду прописать чтобы пакет сам поставился после апгрейда. В любом случае это опциональная фича, стандартные релизы никуда не делись, можно ими пользоваться. Также данный пр не создаёт лишних проблем для мейнейнера: вся сборка делается автоматом. Единственный минус: мне пришлось сделать так чтобы luci app собирался для каждого таргета, но я могу это поправить если хотите, просто пока не заморачивался.

> Вот вы бы лучше сделали аналог fantastic-packages куда поместили бы zapret-openwrt, nfqws-keenetic, youtubeUnblock, podkop, ruantiblock, etc.

Это интересная идея, я над этим подумаю) Но мне кажется не всем нужны сразу все эти пакеты. Имеет смысл сделать гранулярно чтобы можно было добавить только то что нужно. Плюс могут быть конфликты если в разных таких сборных репах будут те же пакеты. К тому же то что репа собирается именно в этом репозитории имеет большой плюс: тут сборка репы идёт автоматом вместе со сборкой пакета. А в отдельном репозитории нужно каким-то образом следить за обновлениями и вовремя запускать сборку, что сложнее

votarum commented:
Поддерживаю. Из CLI как-то привычнее все делать, без таскания файлов. Да и скрипт можно заточить, чтобы архитектура автоматом прописывалась.

Vantuz committed:
Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Update build.yml

Create README

Create README

Update README

Add files via upload

Add files via upload

Update build.yml

Update build.yml

Delete README

clean build.yml

Vantuz committed:
Merge branch 'remittor:master' into master

Vantuz committed:
Merge branch 'remittor:master' into master

Vantuz committed:
fix gpg import

Vantuz committed:
add riscv64_riscv64 arch

Vantuz committed:
Merge pull request #1 from Vantuz/add-riscv

Add riscv

Vantuz committed:
Merge branch 'remittor:master' into master

remittor committed:
config: Fix using autohostlist option into NFQWS_OPT

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	.github/workflows/build.yml

megapro17 commented:
> Обновление пакета сделаю на страничке Zapret. Будет кастомная реализация обновления

Зачем? для чего городить костыльный велосипед? почему вообще нормальной установки и обновления из репозитория нету по умолчанию?

megapro17 commented:
> Я не хочу заставлять пользователя добавлять ссылки на фиды

Вы не хотите заставлять пользователей делать единственный правильный способ установки приложений на openwrt, и вместо этого вынуждает их заниматься какой-то хернёй, по типу ручного закидывания пакетов, которое может один раз будет сделать проще, но нужно повторять постоянно, вместо того чтобы один раз добавить ссылку и удобно устанавливать и обновлять пакет?

> Да и всё равно нужен функционал проверки наличия обновлений. Вот я его объединил с установкой пакетов.

Нет, не нужен! Вас никто не просил лезть в систему и обновлять по своему усмотрению! Вы не пакетный менеджер, это не входит в ваши обязанности!

> Для любителей консоли: /opt/zapret/update-pkg.sh -c -u @

Для любителей консоли вот вам костыльный скрипт который можно сломаться в любой момент, вместо стандартной команды apk upgrade

remittor commented:
> Вас никто не просил лезть в систему и обновлять по своему усмотрению! Вы не пакетный менеджер, это не входит в ваши обязанности!

Вы хоть поняли что написали?
По своему усмотрению я не лезу в систему! И ничего не обновляю по своему усмотрению!
Я пользуюсь услугами стокового пакетного менеджера. Просто это завёрнуто в простой диалог с 1 кнопкой.

remittor commented:
> единственный правильный способ установки приложений на openwrt

Я вот тут уже писал: https://github.com/remittor/zapret-openwrt/pull/193#issuecomment-2842919922
Это мега правильный путь.
Пусть кто-то на github займётся созданием такого. В едином кастом-фиде будут все нужные ipk/apk пакеты.
Можете сами заняться.
